### PR TITLE
Add run strategy, fix ssh key name , fix memory

### PR DESCRIPTION
**Pull request by futuretea**
- Use `run_strategy` instead of `start`
- Read vm memory from limit
- optimize SSH key name display

**Related issues**
https://github.com/harvester/harvester/issues/2225
https://github.com/harvester/harvester/issues/2222

### CODE REVIEW

File: docs/resources/virtualmachine.md [hunk 1]
```diff
@@ -64,12 +64,12 @@ resource "harvester_virtualmachine" "ubuntu20-dev" {
   cpu    = 2
   memory = "2Gi"
 
-  start        = true
+  run_strategy = "Always"
```
Comment (guangbochen) [inline docs/resources/virtualmachine.md:67]: the default run strategy in Harvester is `RerunOnFailure`

File: examples/resources/harvester_virtualmachine/resource.tf [hunk 2]
```diff
@@ -49,12 +49,12 @@ resource "harvester_virtualmachine" "ubuntu20-dev" {
   cpu    = 2
   memory = "2Gi"
 
-  start        = true
+  run_strategy = "Always"
```
Comment (guangbochen) [inline examples/resources/harvester_virtualmachine/resource.tf:52]: ditto

File: internal/provider/virtualmachine/schema_virtualmachine.go [hunk 3]
```diff
@@ -19,10 +23,28 @@ func Schema() map[string]*schema.Schema {
...
 		constants.FieldVirtualMachineStart: {
 			Type:     schema.TypeBool,
 			Optional: true,
-			Default:  true,
+			Deprecated: fmt.Sprintf(`
+please use %s instead of this deprecated field:
+	%s = true  ==>  %s = "%s"
+	%s = false  ==>  %s = "%s"
+`, constants.FieldVirtualMachineRunStrategy,
+				constants.FieldVirtualMachineStart, constants.FieldVirtualMachineRunStrategy, kubevirtv1.RunStrategyAlways,
```
Comment (guangbochen) [inline internal/provider/virtualmachine/schema_virtualmachine.go:46]: ditto

File: docs/resources/virtualmachine.md (hunk 1 above)
Comment (futuretea) [reply docs/resources/virtualmachine.md:67]: If use RerunOnFailure，user can't start/stop vm by terraform

Comment (guangbochen) [reply docs/resources/virtualmachine.md:67]: both `Always` and `RerunOnFailure` will have the same behavior of start and stop, are we using sub-resource action to manage it?
image

Comment (futuretea) [reply docs/resources/virtualmachine.md]: > are we using sub-resource action to manage it?
No, RerunOnFailure  is ok. I thought RerunOnFailure's behavior is like Manual before.

Changed to RerunOnFailure

File: examples/resources/harvester_virtualmachine/resource.tf (hunk 2 above)
Comment (futuretea) [reply examples/resources/harvester_virtualmachine/resource.tf]: @guangbochen updated

**Review by guangbochen — APPROVED** [review]
LGTM

**Pull request merged**